src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -233,10 +236,14 @@ - (void)peripheral:(CBPeripheral *)peripheral
         chip::Ble::ChipBleUUID svcId;
         chip::Ble::ChipBleUUID charId;
         [BleConnection fillServiceWithCharacteristicUuids:characteristic svcId:&svcId charId:&charId];
-        _mBleLayer->HandleWriteConfirmation((__bridge void *) peripheral, &svcId, &charId);
+        dispatch_async(_chipWorkQueue, ^{
```

Comment:
Is this async dispatch of a block with pointers to the stack safe?  This applies in a bunch of places in this file.

Reply:
My understanding of blocks would say it is based on https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Blocks/Articles/bxVariables.html and https://github.com/project-chip/connectedhomeip/blob/4f0ade5fcd50f99f53188ec401e18c2a22f47e14/src/ble/BleUUID.h#L29 but I may have misunderstood something:
```
C++ Objects
In general you can use C++ objects within a block. Within a member function, references to member variables and functions are via an implicitly imported this pointer and thus appear mutable. There are two considerations that apply if a block is copied:

 * If you have a __block storage class for what would have been a stack-based C++ object, then the usual copy constructor is used.
 * If you use any other C++ stack-based object from within a block, it must have a const copy constructor. The C++ object is then copied using that constructor.
```

Reply:
Ah, so `svcId` and `charId` get copied into the block?  OK, that makes sense.